xscen/aggregate.py

```diff
@@ -632,14 +635,19 @@ def spatial_mean(
             # Prepare the History field
             new_history = (
                 f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] "
-                f"xesmf.SpatialAverager over {Path(region['shape']).name} - xESMF v{xe.__version__}"
+                f"xesmf.SpatialAverager over {name} - xESMF v{xe.__version__}"
             )
 
         else:
             raise ValueError("'method' should be one of [bbox, shape].")
 
         kwargs_copy = deepcopy(kwargs)
-        skipna = kwargs_copy.pop("skipna", False)
+        call_kwargs = {"skipna": kwargs_copy.pop("skipna", False)}
+        if "out_chunks" in kwargs:
+            call_kwargs["out_chunks"] = kwargs_copy.pop("out_chunks")
+
+        # Pre-emptive segmentization. Same threshold as xESMF, but there's not strong analysis behind this choice
+        geoms = shapely.segmentize(polygon.geometry, 1)
```

Comment:
Am I right in thinking that this will be skipped if we provided a shapefile with a resolution finer than 1°?

Reply:
It won't do anything to the shape indeed.

`geoms == shapely.segmentize(geoms, 1)` if no geom has any vertex longer than 1.